src/compiler/checker.ts

```diff
@@ -4710,7 +4711,7 @@ namespace ts {
             return undefined;
         }
 
-        function getWidenedTypeFromJSSpecialPropertyDeclarations(symbol: Symbol) {
+        function getWidenedTypeFromJSSpecialPropertyDeclarations(symbol: Symbol, aliased: Symbol) {

```

Comment:
This function's getting pretty big (over 120 lines), any way you could break it down in a future PR?